File: app/src/main/java/mrkm4ntr/twitterclient/sync/TwitterAuthenticatorService.java
package mrkm4ntr.twitterclient.sync;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

public class TwitterAuthenticatorService extends Service {

    private TwitterAuthenticator mAuthenticator;

    @Override
    public void onCreate() {
        mAuthenticator = new TwitterAuthenticator(this);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return mAuthenticator.getIBinder();
    }
}

File: app/src/main/kotlin/mrkm4ntr/twitterclient/sync/TwitterAuthenticatorService.kt
package mrkm4ntr.twitterclient.sync

import android.app.Service
import android.content.Intent
import android.os.IBinder

class TwitterAuthenticatorService : Service() {

    private val authenticator by lazy { TwitterAuthenticator(this) }

    override fun onBind(intent: Intent): IBinder? {
        return authenticator.iBinder
    }
}
